app/objects.js

```diff
@@ -2,14 +2,22 @@ exports = typeof window === 'undefined' ? global : window;
 
 exports.objectsAnswers = {
   alterContext: function(fn, obj) {
-
+    var result = fn.call(obj);
+    return result 
   },
 
   alterObjects: function(constructor, greeting) {
-
+    constructor.prototype.greeting = greeting;
   },
 
   iterate: function(obj) {
 
+    var arr = [];
+
+    for (var key in obj) {
```

Comment:
this can also be done using this cool guy https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys